From 04dcad19c9a709c61781fa1ed701405ef3c5a69d Mon Sep 17 00:00:00 2001
From: Samreen Nilofer <samreen.nilofer@intel.com>
Date: Thu, 12 Feb 2015 10:00:20 +0530
Subject: [PATCH 16/24] ASoC: Intel: platform: Incorporate the Media Mixer
 removal changes

Issue: GMINL-4319

Following changes are made in firmware and accordingly in driver.

1. Media Mixer is removed and the Media mixing functionality is done in SBA
2. Deep Buffer Stream Pipe ID change : Previous :MEDIA3_IN  New : MEDIA2_IN
3. MEDIA3_IN has lost its significance and is not used now
4. PCM2_IN Pipe is added to handle direct Media streams to SBA
5. MMX output pipes (MEDIA0_OUT,MEDIA1_OUT) are removed

Change-Id: Ia770c270506e58c1a1cda6965968dcefa7f71268
Signed-off-by: Hardik T Shah <hardik.t.shah@intel.com>
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/intel/platform-libs/atom_controls.h    | 3 ++-
 sound/soc/intel/platform-libs/atom_pipes.h       | 6 +++---
 sound/soc/intel/platform-libs/controls_v2_dpcm.c | 5 +----
 sound/soc/intel/platform-libs/probe_point_dpcm.c | 4 ----
 4 files changed, 6 insertions(+), 12 deletions(-)

diff --git a/sound/soc/intel/platform-libs/atom_controls.h b/sound/soc/intel/platform-libs/atom_controls.h
index 8dea361..1d813b7 100644
--- a/sound/soc/intel/platform-libs/atom_controls.h
+++ b/sound/soc/intel/platform-libs/atom_controls.h
@@ -93,8 +93,9 @@
 #define SST_IP_MEDIA1		SST_MIX_IP(18)
 #define SST_IP_MEDIA2		SST_MIX_IP(19)
 #define SST_IP_MEDIA3		SST_MIX_IP(20)
+#define SST_IP_PCM2		SST_MIX_IP(21)
 
-#define SST_IP_LAST		SST_IP_MEDIA3
+#define SST_IP_LAST		SST_IP_PCM2
 
 #define SST_SWM_INPUT_COUNT	(SST_IP_LAST + 1)
 #define SST_CMD_SWM_MAX_INPUTS	6
diff --git a/sound/soc/intel/platform-libs/atom_pipes.h b/sound/soc/intel/platform-libs/atom_pipes.h
index a4e96d7..47178a0 100644
--- a/sound/soc/intel/platform-libs/atom_pipes.h
+++ b/sound/soc/intel/platform-libs/atom_pipes.h
@@ -93,6 +93,7 @@ enum sst_dfw_path_index {
 
 	SST_DFW_PATH_INDEX_PCM0_IN	      = (0x8D << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_PCM1_IN	      = (0x8E << SST_DFW_PATH_ID_SHIFT),
+	SST_DFW_PATH_INDEX_PCM2_IN	      = (0x9E << SST_DFW_PATH_ID_SHIFT),
 
 	SST_DFW_PATH_INDEX_MEDIA0_IN	      = (0x8F << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_MEDIA1_IN	      = (0x90 << SST_DFW_PATH_ID_SHIFT),
@@ -109,7 +110,6 @@ enum sst_dfw_path_index {
 	SST_DFW_PATH_INDEX_PROBE7_PIPE_IN     = (0x99 << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_PROBE8_PIPE_IN     = (0x9A << SST_DFW_PATH_ID_SHIFT),
 
-	SST_DFW_PATH_INDEX_MEDIA3_IN	      = (0x9C << SST_DFW_PATH_ID_SHIFT),
 	SST_DFW_PATH_INDEX_LOW_PCM0_IN	      = (0x9D << SST_DFW_PATH_ID_SHIFT),
 
 	SST_DFW_PATH_INDEX_RESERVED	      = (0xFF << SST_DFW_PATH_ID_SHIFT),
@@ -149,6 +149,8 @@ enum sst_dfw_swm_inputs {
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_PCM1		= (SST_DFW_PATH_INDEX_PCM1_IN	     |
 						SST_DFW_DEFAULT_CELL_NBR),
+	SST_DFW_SWM_IN_PCM2		= (SST_DFW_PATH_INDEX_PCM2_IN	     |
+						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_MEDIA0		= (SST_DFW_PATH_INDEX_MEDIA0_IN	     |
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_MEDIA1		= (SST_DFW_PATH_INDEX_MEDIA1_IN	     |
@@ -157,8 +159,6 @@ enum sst_dfw_swm_inputs {
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_FM		= (SST_DFW_PATH_INDEX_FM_IN	     |
 						SST_DFW_DEFAULT_CELL_NBR),
-	SST_DFW_SWM_IN_MEDIA3		= (SST_DFW_PATH_INDEX_MEDIA3_IN	     |
-						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_LOW_PCM0		= (SST_DFW_PATH_INDEX_LOW_PCM0_IN    |
 						SST_DFW_DEFAULT_CELL_NBR),
 	SST_DFW_SWM_IN_END		= (SST_DFW_PATH_INDEX_RESERVED	     |
diff --git a/sound/soc/intel/platform-libs/controls_v2_dpcm.c b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
index 76887e1..9453864 100644
--- a/sound/soc/intel/platform-libs/controls_v2_dpcm.c
+++ b/sound/soc/intel/platform-libs/controls_v2_dpcm.c
@@ -787,12 +787,9 @@ static const uint swm_dfw_mixer_input_ids[] = {
 	[SST_IP_VOIP]		= SST_DFW_SWM_IN_VOIP,
 	[SST_IP_PCM0]		= SST_DFW_SWM_IN_PCM0,
 	[SST_IP_PCM1]		= SST_DFW_SWM_IN_PCM1,
+	[SST_IP_PCM2]		= SST_DFW_SWM_IN_PCM2,
 	[SST_IP_LOW_PCM0]	= SST_DFW_SWM_IN_LOW_PCM0,
 	[SST_IP_FM]		= SST_DFW_SWM_IN_FM,
-	[SST_IP_MEDIA0]		= SST_DFW_SWM_IN_MEDIA0,
-	[SST_IP_MEDIA1]		= SST_DFW_SWM_IN_MEDIA1,
-	[SST_IP_MEDIA2]		= SST_DFW_SWM_IN_MEDIA2,
-	[SST_IP_MEDIA3]		= SST_DFW_SWM_IN_MEDIA3,
 };
 
 /* Look up table to convert MIXER SW bit regs to SWM inputs */
diff --git a/sound/soc/intel/platform-libs/probe_point_dpcm.c b/sound/soc/intel/platform-libs/probe_point_dpcm.c
index bff107d..faaea12 100644
--- a/sound/soc/intel/platform-libs/probe_point_dpcm.c
+++ b/sound/soc/intel/platform-libs/probe_point_dpcm.c
@@ -19,10 +19,6 @@
 static const struct sst_probe_config sst_probes[] = {
 	/* TODO: get this struct from FW config data */
 	/* gain outputs  */
-	{ "media0_in gain", SST_PATH_INDEX_MEDIA0_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_MMX, { 1, 2, 1 } },
-	{ "media1_in gain", SST_PATH_INDEX_MEDIA1_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_MMX, { 1, 2, 1 } },
-	{ "media2_in gain", SST_PATH_INDEX_MEDIA2_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_MMX, { 1, 2, 1 } },
-	{ "media3_in gain", SST_PATH_INDEX_MEDIA3_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_MMX, { 1, 2, 1 } },
 	{ "pcm0_in gain", SST_PATH_INDEX_PCM0_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_SBA, { 1, 2, 1 } },
 	{ "pcm1_in gain", SST_PATH_INDEX_PCM1_IN, SST_MODULE_ID_GAIN_CELL, SST_TASK_SBA, { 1, 2, 1 } },
 	{ "pcm1_out gain", SST_PATH_INDEX_PCM1_OUT, SST_MODULE_ID_GAIN_CELL, SST_TASK_SBA, { 1, 2, 1 } },
-- 
1.9.1

